src/lib.rs

```diff
@@ -1770,4 +1842,49 @@ mod tests {
 			assert!(dt_to_generalized(&dt).is_ok());
 		}
 	}
+
+	#[test]
+	fn test_with_key_usages() {
```

Comment:
Can you put this test into `tests/generic.rs`?

Also it would be great if there were at least one test with webpki.

Reply:
Wait no it's ok, I think this entire module should be moved to the tests directory but I'll do that in a later change. It can stay in here.

Reply:
I added another test to test coverage with bytes > 8.